include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,357 @@
+     * // and give the cell data vector that we intend to unpack later,
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell-wise values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_for_coarsening_and_refinement(data_to_transfer);
+     *
+     * // actually execute the refinement,
+     * triangulation.execute_coarsening_and_refinement();
+     *
+     * // unpack transferred data,
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.unpack(transferred_data);
+     *
+     * @endcode
+     *
+     *
+     * <h3>Use for serialization</h3>
+     *
+     * This class can be used to serialize and later deserialize a distributed
+     * mesh with attached data to separate files.
+     *
+     * For serialization, the following code snippet saves not only the
+     * triangulation itself, but also the cell-wise data attached:
+     * @code
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell-wise values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_for_serialization(data_to_transfer);
+     *
+     * triangulation.save(filename);
+     * @endcode
+     *
+     * Later, during deserialization, both triangulation and data can be
+     * restored as follows:
+     * @code
+     * //[create coarse mesh...]
+     * triangulation.load(filename);
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.deserialize(transferred_data);
+     * @endcode
+     *
+     * @note If you use more than one object to transfer data via the
+     * parallel::distributed::Triangulation::register_data_attach() and
+     * parallel::distributed::Triangulation::notify_ready_for_unpack() interface
+     * with the aim of serialization, the calls to the corresponding
+     * prepare_for_serialization() and deserialize() functions need to happen in
+     * the same order, respectively. Classes relying on this interface are e.g.
+     * parallel::distributed::CellDataTransfer,
+     * parallel::distributed::SolutionTransfer, and Particles::ParticleHandler.
+     *
+     * @note See the documentation of parallel::distributed::SolutionTransfer for
+     * matching code snippets for both transfer and serialization.
+     *
+     * @ingroup distributed
+     * @author Marc Fehling, 2018
+     */
+    template <int dim, int spacedim = dim, typename VectorType = Vector<double>>
+    class CellDataTransfer
+    {
+    public:
+      /**
+       * Whenever data shall be transferred across coarsening, it is not a
```

Comment:
maybe something like: 

When the date is transferred during coarsening, it is not trivial to decide how to process the data of child cells which will be coarsened.

?